code/__HELPERS/icons.dm

```diff
@@ -761,13 +758,15 @@ world
 	var/curicon = A.icon || deficon
 	var/curstate = A.icon_state || defstate
 
-	if(!((noIcon = (!curicon))))
+	var/render_icon = curicon
+
+	if (render_icon)
 		var/curstates = icon_states(curicon)
 		if(!(curstate in curstates))
-			if("" in curstates)
+			if ("" in curstates)
 				curstate = ""
 			else
-				noIcon = TRUE // Do not render this object.
+				render_icon = FALSE
 
 	var/curdir
```

Comment:
itd be better if all of the cur[x] vars were declared in the same place

Reply:
It's not with the rest of them since it's only declared right after here.

Reply:
Though I'm now realizing I can make it a ternary and get the best of both world.